crate_universe/src/utils/starlark/select_dict.rs

```diff
@@ -0,0 +1,513 @@
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::Debug;
+
+use serde::ser::{SerializeMap, Serializer};
+use serde::Serialize;
+use serde_starlark::{FunctionCall, MULTILINE};
+
+use crate::select::{Select, SelectableValue};
+use crate::utils::starlark::{
+    looks_like_bazel_configuration_label, NoMatchingPlatformTriples, WithOriginalConfigurations,
+};
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct SelectDict<T>
+where
+    T: SelectableValue,
+{
+    // Invariant: keys in this map are not in any of the inner maps of `selects`.
+    common: BTreeMap<String, T>,
+    // Invariant: none of the inner maps are empty.
+    selects: BTreeMap<String, BTreeMap<String, WithOriginalConfigurations<T>>>,
```

Comment:
They are a `Label` at this point, so that's a good suggestion.

Likewise `platforms: &BTreeMap<String, BTreeSet<String>>` should presumably be `platforms: &BTreeMap<String, BTreeSet<Label>>`.  I think...  Which reinforces your point 😆